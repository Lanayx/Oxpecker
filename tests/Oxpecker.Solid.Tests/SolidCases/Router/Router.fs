module Oxpecker.Solid.Tests.Cases.Router

open Oxpecker.Solid
open Oxpecker.Solid.Router
open Oxpecker.Solid.Tests.SolidCases.Router.App1
open Oxpecker.Solid.Tests.SolidCases.Router.App2

[<SolidComponent>]
let Root (props: RootProps) : HtmlElement =
    Fragment() {
        h1() { "Root header" }
        props.children
        A(href="/about") { "About" }
    }

[<SolidComponent>]
let Test1 () =
    Router(root=Root) {
        Route()
        Route(path="/def") {
            Route(path="/inner", component'=App1)
        }
        Route(path="/ghi", component'=App2)
    }

[<SolidComponent>]
let Test2() =
    let routes = [|
        RootConfig("/app1", lazy'(fun () -> importComponent("/App1")))
        RootConfig("/app2", lazy'(fun () -> importComponent("/App2")))
    |]

    Router() {
        routes
    }
