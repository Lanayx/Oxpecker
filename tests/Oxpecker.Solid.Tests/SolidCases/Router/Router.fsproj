<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net8.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <Compile Include="App1.fs" />
    <Compile Include="App2.fs" />
    <Compile Include="Router.fs" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\..\..\..\src\Oxpecker.Solid\Oxpecker.Solid.fsproj" />
  </ItemGroup>

</Project>
