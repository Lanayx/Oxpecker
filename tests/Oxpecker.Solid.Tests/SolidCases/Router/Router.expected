import { Route, Router } from "@solidjs/router";
import { App1 } from "./App1.fs.js";
import { App2 } from "./App2.fs.js";
import { lazy } from "solid-js";

export function Root(props) {
    return <>
        <h1>
            Root header
        </h1>
        {props.children}
    </>;
}

export function Test1() {
    return <Router root={Root}>
        <Route></Route>
        <Route path="/def">
            <Route path="/inner"
                component={App1}></Route>
        </Route>
        <Route path="/ghi"
            component={App2}></Route>
    </Router>;
}

export function Test2() {
    const routes = [{
        path: "/app1",
        component: lazy(() => (import("/App1"))),
    }, {
        path: "/app2",
        component: lazy(() => (import("/App2"))),
    }];
    return <Router>
        {routes}
    </Router>;
}

